warehouse/query-core/src/main/java/datawave/query/transformer/DocumentTransformer.java

```diff
@@ -165,80 +165,90 @@ static protected Key correctKey(Key origKey) {
+            return _transform(documentEntry);
+        } else {
+            return null;
+        }
+    }
+    
+    @Override
+    public Object transform(Object input) throws EmptyObjectException {
         if (input instanceof Entry<?,?>) {
             @SuppressWarnings("unchecked")
             Entry<Key,org.apache.accumulo.core.data.Value> entry = (Entry<Key,org.apache.accumulo.core.data.Value>) input;
             
             Entry<Key,Document> documentEntry = deserializer.apply(entry);
-            
-            documentKey = correctKey(documentEntry.getKey());
-            document = documentEntry.getValue();
-            
-            if (null == documentKey || null == document)
-                throw new IllegalArgumentException("Null key or value. Key:" + documentKey + ", Value: " + entry.getValue());
-            
-            extractMetrics(document, documentKey);
-            document.debugDocumentSize(documentKey);
-            
-            String row = documentKey.getRow().toString();
-            
-            String colf = documentKey.getColumnFamily().toString();
-            
-            int index = colf.indexOf("\0");
-            Preconditions.checkArgument(-1 != index);
-            
-            dataType = colf.substring(0, index);
-            uid = colf.substring(index + 1);
-            
-            // We don't have to consult the Document to rebuild the Visibility, the key
-            // should have the correct top-level visibility
-            ColumnVisibility eventCV = new ColumnVisibility(documentKey.getColumnVisibility());
-            
-            try {
-                
-                for (String contentFieldName : this.contentFieldNames) {
-                    if (document.containsKey(contentFieldName)) {
-                        Attribute<?> contentField = document.remove(contentFieldName);
-                        if (contentField.getData().toString().equalsIgnoreCase("true")) {
-                            Content c = new Content(uid, contentField.getMetadata(), document.isToKeep());
-                            document.put(contentFieldName, c, false, this.reducedResponse);
-                        }
-                    }
-                }
-                
-                for (String primaryField : this.primaryToSecondaryFieldMap.keySet()) {
-                    if (!document.containsKey(primaryField)) {
-                        for (String secondaryField : this.primaryToSecondaryFieldMap.get(primaryField)) {
-                            if (document.containsKey(secondaryField)) {
-                                document.put(primaryField, document.get(secondaryField), false, this.reducedResponse);
-                                break;
-                            }
-                        }
-                    }
+            for (DocumentTransform transform : transforms) {
+                if (documentEntry != null) {
+                    documentEntry = transform.apply(documentEntry);
```

Comment:
If the grouping transformer runs first here, and it returns null (which it always does), the unique transformer will never run.

Reply:
Ok, I get it.  This isn't a problem right now, but if any transforms are added in the future, they'll have to be added prior to this one.